MGT Person - Disable attempt to get user photo when using userId/personQuery and fetchImage=false

NathZ1 opened the pull request:
Closes #1558 

### PR Type
Bugfix

### Description of the changes
Fixes bug #1558 by adding additional checks (mainly checking the fetchImage prop) prior to running getUserWithPhoto() / getPersonImage() functions.

NathZ1 commented on packages/mgt-components/src/components/mgt-person/mgt-person.ts:
```diff
@@ -1019,7 +1019,12 @@ export class MgtPerson extends MgtTemplatedComponent {
-      if (this._avatarType === 'photo') {
+      if (
+        this._avatarType === 'photo' &&
+        this.fetchImage &&
```
Can easily make that change, but that checking logic was already present a tad higher up (around line 1002) which doesn't have an undefined check and I can't remember seeing it fail previously?

```
if (details) {
      if (
        !details.personImage &&
        this.fetchImage &&
        this._avatarType === 'photo' &&
        !this.personImage &&
        !this._fetchedImage
      ) {
        details;
        let image;
        if ('groupTypes' in details) {
          image = await getGroupImage(graph, details, MgtPerson.config.useContactApis);
        } else {
          image = await getPersonImage(graph, details, MgtPerson.config.useContactApis);
        }
        if (image) {
          details.personImage = image;
          this._fetchedImage = image;
        }
      }
```

NathZ1 replied:
added undefined check :)

NathZ1 commented:
@sebastienlevert when i run yarn prettier:write it decides to remove line 1019 for some reason (which is surely not a good thing) and formats the ifs to a single line (which personally i don't mind, however that is different to the existing if on lines 1000-1006 above which was already split across multiple lines). Not sure how you'd like me to proceed here...

![image](https://user-images.githubusercontent.com/3673816/154782760-6e4de4ed-dc16-405a-98e3-d0b009e3a485.png)

Mnickii commented on packages/mgt-components/src/components/mgt-person/mgt-person.ts:
```diff
@@ -1040,7 +1045,12 @@ export class MgtPerson extends MgtTemplatedComponent {
         this.personDetailsInternal = people[0];
-        if (this._avatarType === 'photo') {
+        if (
+          this._avatarType === 'photo' &&
+          this.fetchImage &&
+          !this.personImage &&
+          !this._fetchedImage
```
> We have this block repeated above, could we have it as a reusable var?

NathZ1 replied:
added imageNeeded var

NathZ1 commented:
> It seems `prettier` is failing on your code @NathZ1. Can I ask you to just run `yarn prettier:write` to ensure code styling is respected? Thanks!

Passing now :)

sebastienlevert commented:
Hey @NathZ1! After discussions with the team, this seems to clash with other uses of the `fetch-image` property in other areas of the code. So here is what we would be suggesting. Instead of reusing the `fetch-image` property, adding a new `disable-image-fetch` would enable this specific scenario without breaking the rest of the library. These properties are not "exactly" real boolean properties as they are used as flags. If it's present, it's `true`, if not, it's `false` which makes the `image-fetch="false"` scenario unreliable.

The `image-fetch` is really meant to be used when we pass in a `personDetails` but we want to "force" the fetch of the image because we don't already have it.

If this makes sense to you, you can absolutely go ahead and make the few changes and this should be good for review! Thanks!

NathZ1 commented:
Hi @sebastienlevert, i've updated the PR to instead add a new property. Added story and created PR to microsoft-graph-docs as well (https://github.com/microsoftgraph/microsoft-graph-docs/pull/16046)

musale reviewed (approved):
This looks good to me. Storybook will not build on PR because it's a PR from a fork.

musale reviewed (approved):
Works on Edge, Chrome and Firefox.

Mnickii reviewed (approved):
Looks good to me. Tested on Chrome, Firefox and Edge.
